### PR TITLE
fix-bug-558: adding column left or right unpredictable state

### PR DESCRIPTION
**Pull request by gibsonliketheguitar**
#558 
- applying the single responsibility principle, to improve maintenance in the future.  
- Introduce new table action called 'insert' to replace 'handleSave' in NewColumn Component

## Reason

handleModalSave  === tableActions.column.update === updateColumn

```ts
 const menuModalProps = {
   ...
    handleSave: handleModalSave,
  };

const handleModalSave = (key: string, update: Record<string, any>) => {
    actions.update(key, update);
};
 
const actions: TableActions = { 
...
column: {
      ...
      update: configActions.updateColumn,
    },
}

const updateColumn = (key: string, updates: any) => {
    const { columns } = tableConfigState;

    const updatedColumns = {
      ...columns,                                   // this maintains previous index
      [key]: { ...columns[key], ...updates },       // this allows for duplicate index
    };
  };
```

Bug
- a copy of a new column can result in duplicated indexes
-  after copy, the other column's index has to be updated 

Solution
- calculate the insert position, given left or right position,  and handle the edge case
- use _orderBy lodash to covert  columns into a data structure where we can use .splice on
- after insert, use reduce to rewrite index and return data into object format for dispatcher function

### CODE REVIEW

File: src/hooks/useTable/useTableConfig.ts [hunk 1]
```diff
@@ -88,6 +89,8 @@ const useTableConfig = (tablePath?: string) => {
     column.width = width;
     let updatedColumns = columns;
     updatedColumns[column.key] = column;
+
+    console.log("what is updatedColumns resiz3", updatedColumns);
```
Comment (gibsonliketheguitar) [inline src/hooks/useTable/useTableConfig.ts:93]: remove console.log

File: src/hooks/useTable/useTableConfig.ts [hunk 2]
```diff
@@ -112,6 +115,32 @@ const useTableConfig = (tablePath?: string) => {
       data: { columns: updatedColumns },
     });
   };
+
+  /** insert column by index
+   * @param col     properties of new column
+   * @param source  source object { index: selected index, position: left | right }
+   */
+
+  const insert = (col, source) => {
```
Comment (gibsonliketheguitar) [inline src/hooks/useTable/useTableConfig.ts:124]: double check col insert, with the last column header

File: src/hooks/useTable/useTableConfig.ts [hunk 3]
```diff
@@ -112,6 +115,32 @@ const useTableConfig = (tablePath?: string) => {
...
+    const { columns } = tableConfigState;
+    const position = source.position === "right" ? 1 : 0;
+    const insertPos = source.index + position;
+    const targetPos = insertPos === -1 ? 0 : insertPos;
```
Comment (gibsonliketheguitar) [inline src/hooks/useTable/useTableConfig.ts:128]: document logic here

**Comment by gibsonliketheguitar** [comment]
wait for #626 to be merge. 626 contains fix for keeping the index correct. 
This feature breaks when columns are delete because index are not updated

**Review by shamsmosowi — APPROVED** [review]

**Pull request merged**